src/main/java/org/apache/sling/models/impl/ReflectionUtil.java

```diff
@@ -34,6 +34,18 @@
  */
 public final class ReflectionUtil {
 
+    static Class<?> recordType;
+
+    static {
+        try {
+            recordType = Class.forName("java.lang.Record");
+        } catch (ClassNotFoundException e) {
+            // this happens when running with Java11, which is supported, but
+            // of course does not have support for Record types
+            recordType = null;
```

Comment:
IMO, It would be beneficial to include a log indicating whether Records are supported in the current setup to enhance visibility and traceability. 

_Additionally nice to have would be, refactoring the supported JDK feature/classes logic into a separate class and implementing caching to minimize repeated null checks and isAssignableFrom calls during serialization, especially when handling large numbers of objects, **could be** a useful performance optimization. (but can be ignored if we are not facing significant performance degradation :) )_

Reply:
> Additionally nice to have would be, refactoring the supported JDK feature/classes logic into a separate class and implementing caching to minimize repeated null checks and isAssignableFrom calls during serialization, especially when handling large numbers of objects, could be a useful performance optimization. (but can be ignored if we are not facing significant performance degradation :) )

I have started that in [0a391ba](https://github.com/apache/sling-org-apache-sling-models-impl/pull/61/commits/0a391ba113e87d62579986bc5f4a30bd37985d49).